cpp/include/Ice/ServerAuthenticationOptions.h

```diff
@@ -0,0 +1,415 @@
+         * callback is set, it replaces the default validation callback and must perform all necessary validation
+         * steps. If trustedRootCertificates is set, the passed context object will use them as the trusted root
+         * certificates for validation. This setting can be modified by the application using
+         * [SecTrustSetAnchorCertificates](https://developer.apple.com/documentation/security/1396098-sectrustsetanchorcertificates?language=objc)
+         *
+         * Example of setting clientCertificateValidationCallback:
+         * ```cpp
+         * communicator->createObjectAdapterWithEndpoints(
+         *   "Hello",
+         *   "ssl -h 127.0.0.1 -p 10000",
+         *   ServerAuthenticationOptions {
+         *     .clientCertificateValidationCallback =
+         *       [](CtxtHandle context, const IceSSL::ConnectionInfoPtr& info) {
+         *          ...
+         *       }
+         * });
+         * ```
+         *
+         * @param context A CtxtHandle representing the security context associated with the current connection. This
+         * context contains security data relevant for validation, such as the client's certificate chain and cipher
+         * suite.
+         * @param info The IceSSL::ConnectionInfoPtr object that provides additional connection-related data which might
+         * be relevant for contextual certificate validation.
+         * @return true if the certificate chain is valid and the connection should proceed; false if the certificate
+         * chain is invalid and the connection should be aborted.
+         * @throws Ice::SecurityException if the certificate chain is invalid and the connection should be aborted.
+         *
+         * [See
+         * SecTrustEvaluateWithError](https://developer.apple.com/documentation/security/2980705-sectrustevaluatewitherror?language=objc)
+         */
+        std::function<bool(CtxtHandle context, const IceSSL::ConnectionInfoPtr& info)>
+            clientCertificateValidationCallback;
+#elif defined(ICE_USE_SECURE_TRANSPORT)
+        /**
+         * A callback that allows selecting the server's SSL certificate chain based on the name of the object adapter
+         * that accepts the connection.
+         *
+         * @remarks This callback is invoked by the SSL transport for each new incoming connection before starting the
+         * SSL handshake to determine the appropriate server certificate chain. The callback should return a CFArrayRef
+         * that represents the server's certificate chain, or NULL if no certificate chain should be used for the
+         * connection. The SSL transport takes ownership of the returned CFArrayRef and releases it when the connection
+         * is closed.
+         *
+         * @param adapterName The name of the object adapter that accepted the connection.
+         * @return CFArrayRef containing the server's certificate chain, or NULL to indicate that no certificate is
```

Comment:
Should we still prefer `nullptr` to `NULL` here?

Reply:
`nullptr`